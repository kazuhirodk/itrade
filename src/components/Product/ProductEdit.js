import React, { Component } from 'react';
import { Actions } from 'react-native-router-flux';
import firebase from 'react-native-firebase'
import FirebaseService from '../../../services/FirebaseService';
import {
  StyleSheet,
  Text,
  View,
  Image,
  TextInput,
  Button
} from 'react-native';

const goToHome = () => {
  Actions.home()
}

key = '';
user = '';

export default class ProductEdit extends Component {
  constructor (props) {
    super(props)
    this.productId = this.props.productEditId;
    this.state = {
      nome: '',
      preco: '',
      descricao: '',
      foto: 'imageUrl'
    }
  }

  componentDidMount() {
    firebase.auth().onAuthStateChanged(userLogged => {
      const list = FirebaseService.getDataList('usuarios', function(){});

      list.orderByChild("email").equalTo(userLogged.email).on("child_added", snapshot => {
        key = snapshot.key;
        user = snapshot.val();
        product = user.produtos[this.productId];

        this.setState({
          nome: product.nome,
          preco: product.preco,
          descricao: product.descricao,
          foto: product.foto
        })
      });
    })
  }

  onChangeText = (key, val) => {
    this.setState({ [key]: val })
  }

  updateProduct = async() => {
    try {
      FirebaseService.updateData('usuarios/' + key + '/produtos/' + this.productId, this.state);
      alert("Atualização realizada com sucesso!")
    } catch(e){
      alert("Atualização falhou.")
    }
    goToHome();
  }

  render() {
    return (
      <View style={styles.container}>
        <View style={styles.header}></View>
        <Image style={styles.avatar} source={{uri: this.state.foto}}/>
        <View style={styles.body}>
          <Text style={styles.name}>{this.state.nome}</Text>
          <Text style={styles.info}>Preço médio: R$ {this.state.preco}</Text>
          <Text style={styles.description}>{this.state.descricao}</Text>
        </View>
        <TextInput
          style={styles.input}
          placeholder='Nome'
          autoCapitalize="none"
          placeholderTextColor='#CD7F32'
          onChangeText={val => this.onChangeText('nome', val)}
          value={this.state.nome}
        />
        <TextInput
          style={styles.input}
          placeholder='Preço'
          autoCapitalize="none"
          placeholderTextColor='#CD7F32'
          onChangeText={val => this.onChangeText('preco', val)}
          value={this.state.preco.toString()}
        />
        <TextInput
          style={styles.input}
          placeholder='Nova Descrição'
          autoCapitalize="none"
          placeholderTextColor='#CD7F32'
          onChangeText={val => this.onChangeText('descricao', val)}
          value={this.state.descricao}
        />
        <Button
          color='#239033'
          title='Salvar'
          onPress={this.updateProduct}
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    padding: 20,
    flex: 1,
    backgroundColor: '#FFFAFA',
    alignContent: 'center'
  },
  header:{
    backgroundColor: "#CD7F32",
    height:200,
  },
  avatar: {
    width: 130,
    height: 130,
    borderRadius: 63,
    borderWidth: 4,
    borderColor: "white",
    marginBottom:10,
    alignSelf:'center',
    position: 'absolute',
    marginTop:130
  },
  body:{
    marginTop: 40,
    alignItems: 'center',
    marginBottom: 40
  },
  name:{
    fontSize:28,
    color: "#696969",
    fontWeight: "600"
  },
  info:{
    fontSize:16,
    color: "#CD7F32",
    marginTop:10
  },
  description:{
    fontSize:16,
    color: "#696969",
    marginTop:10,
    textAlign: 'center'
  },
  input: {
    height: 40,
    backgroundColor: '#E3E1E1',
    marginBottom: 10,
    padding: 10,
    color: '#000'
  },
});
