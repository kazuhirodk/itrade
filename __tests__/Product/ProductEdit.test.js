import ProductEdit from '../../src/components/Product/ProductEdit.js'
import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import renderer from 'react-test-renderer';

Enzyme.configure({ adapter: new Adapter() });

describe('#ProductEdit', () => {
  it('matches snapshot properly', () => {
    const jsonProductEdit = renderer.create(<ProductEdit />).toJSON();
    expect(jsonProductEdit).toMatchSnapshot();
  });
});
