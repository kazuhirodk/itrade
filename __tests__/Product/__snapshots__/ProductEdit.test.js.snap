// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#ProductEdit matchs snapshot properly 1`] = `
<View
  style={
    Object {
      "alignContent": "center",
      "backgroundColor": "#FFFAFA",
      "flex": 1,
      "padding": 20,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#CD7F32",
        "height": 200,
      }
    }
  />
  <Image
    source={
      Object {
        "testUri": "../../../src/components/images/product1.png",
      }
    }
    style={
      Object {
        "alignSelf": "center",
        "borderColor": "white",
        "borderRadius": 63,
        "borderWidth": 4,
        "height": 130,
        "marginBottom": 10,
        "marginTop": 130,
        "position": "absolute",
        "width": 130,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "marginBottom": 40,
        "marginTop": 40,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#696969",
          "fontSize": 28,
          "fontWeight": "600",
        }
      }
    >
      Product Name
    </Text>
    <Text
      style={
        Object {
          "color": "#CD7F32",
          "fontSize": 16,
          "marginTop": 10,
        }
      }
    >
      Preço médio: R$ 97,00
    </Text>
    <Text
      style={
        Object {
          "color": "#696969",
          "fontSize": 16,
          "marginTop": 10,
          "textAlign": "center",
        }
      }
    >
      Lorem ipsum dolor sit amet, saepe sapientem eu nam. Qui ne assum electram expetendis, omittam deseruisse consequuntur ius an
    </Text>
  </View>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    placeholder="Nome"
    placeholderTextColor="#CD7F32"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "#E3E1E1",
        "color": "#000",
        "height": 40,
        "marginBottom": 10,
        "padding": 10,
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    placeholder="Preço"
    placeholderTextColor="#CD7F32"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "#E3E1E1",
        "color": "#000",
        "height": 40,
        "marginBottom": 10,
        "padding": 10,
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    placeholder="Nova Descrição"
    placeholderTextColor="#CD7F32"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "#E3E1E1",
        "color": "#000",
        "height": 40,
        "marginBottom": 10,
        "padding": 10,
      }
    }
    underlineColorAndroid="transparent"
  />
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    clickable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#239033",
            },
          ]
        }
      >
        Salvar
      </Text>
    </View>
  </View>
</View>
`;
